feat: add snowflake integration doc (#1540)
SigNoz/signoz.io: data/docs/integrations/snowflake.mdx

  # otlp/snowflake:
  #   endpoint: "localhost:4317"
  #   tls:
  #     insecure: true

service:
  pipelines:
    metrics/snowflake:
      receivers: [snowflake]
      processors: [resourcedetection/system]
      exporters: [otlp/snowflake]
```

All the available Snowflake receiver configurations can be found [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/snowflakereceiver/config.go).

### Step 2: Set Environment Variables

Set the following environment variables:

```bash

# Snowflake account from where the metrics need to be collected
export SNOWFLAKE_ACCOUNT="<organization>-<account-name>"

# Snowflake username used while logging in into the Snowflake account
export SNOWFLAKE_USERNAME="<snowflake-username>"

# Password for the corresponding Snowflake user used while logging in into the Snowflake account
export SNOWFLAKE_PASSWORD="<snowflake-password>"

# Snowflake warehouse
export SNOWFLAKE_WAREHOUSE="<snowflake-warehouse>"

# region specific SigNoz cloud ingestion endpoint
export OTLP_DESTINATION_ENDPOINT="ingest.<region>.signoz.cloud:443"

# your SigNoz ingestion key
export SIGNOZ_INGESTION_KEY="<your-ingestion-key>"
```

> - Set the <region> to match your SigNoz Cloud [region](https://signoz.io/docs/ingestion/signoz-cloud/overview/#endpoint)

Reviewer: ```suggestion
- Set the `<region>` to match your SigNoz Cloud [region](https://signoz.io/docs/ingestion/signoz-cloud/overview/#endpoint)
```

> - Replace <your-ingestion-key> with your SigNoz [ingestion key](https://signoz.io/docs/ingestion/signoz-cloud/keys/)

Reviewer: ```suggestion
- Replace `<your-ingestion-key>` with your SigNoz [ingestion key](https://signoz.io/docs/ingestion/signoz-cloud/keys/)
```

---
date: 2024-08-03
id: snowflake
title: Snowflake Metrics
---

## Overview

This integration helps you to monitor key Snowflake metrics, and view them with an out-of-the-box dashboard.

## Prerequisites

Before you begin, ensure you have:

1. A Snowflake account
2. An OpenTelemetry (OTEL) Collector with access to Snowflake
   - [Install the OTEL Collector](https://signoz.io/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/)(v0.88.0 or newer) if not done already 
   - Ensure you can provide config files and set environment variables for the collector

>    - Ensure the OTEL collector has access to Snowflake for metrics collection

Reviewer: @shrutimantri this should be the first step for setup.

Author: Removing the third point. Snowflake can always be connected from any host as long as the username and password are correct.

> ## Collecting Snowflake Metrics

Reviewer: ```suggestion
## Setup
```

    tls:
      insecure: false
    headers:
      "signoz-ingestion-key": "${env:SIGNOZ_INGESTION_KEY}"

  # export to local collector
  # otlp/snowflake:
  #   endpoint: "localhost:4317"
  #   tls:
  #     insecure: true

service:
  pipelines:
    metrics/snowflake:
      receivers: [snowflake]
      processors: [resourcedetection/system]
      exporters: [otlp/snowflake]
```

All the available Snowflake receiver configurations can be found [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/snowflakereceiver/config.go).

### Step 2: Set Environment Variables

Set the following environment variables:

```bash

# Snowflake account from where the metrics need to be collected
export SNOWFLAKE_ACCOUNT="<organization>-<account-name>"

# Snowflake username used while logging in into the Snowflake account
export SNOWFLAKE_USERNAME="<snowflake-username>"

# Password for the corresponding Snowflake user used while logging in into the Snowflake account
export SNOWFLAKE_PASSWORD="<snowflake-password>"

# Snowflake warehouse
export SNOWFLAKE_WAREHOUSE="<snowflake-warehouse>"

# region specific SigNoz cloud ingestion endpoint

> export OTLP_DESTINATION_ENDPOINT="ingest.{REGION}.signoz.cloud:443"

Reviewer: ```suggestion
export OTLP_DESTINATION_ENDPOINT="ingest.<region>.signoz.cloud:443"
```

# your SigNoz ingestion key

> export SIGNOZ_INGESTION_KEY="signoz-ingestion-key"

Reviewer: ```suggestion
export SIGNOZ_INGESTION_KEY="<your-ingestion-key>"
```

```

> You can find more details about SigNoz ingestion keys and regions [here](https://signoz.io/docs/ingestion/signoz-cloud/overview/)

Reviewer: ```suggestion
- Set the <region> to match your SigNoz Cloud [region](https://signoz.io/docs/ingestion/signoz-cloud/overview/#endpoint)
- Replace <your-ingestion-key> with your SigNoz [ingestion key](https://signoz.io/docs/ingestion/signoz-cloud/keys/)
```

### Step 3: Use the Collector Config File

Add the following flag to your collector run command:
```bash
--config snowflake-metrics-collection-config.yaml
```

> **Note:** The collector can use multiple config files by specifying multiple `--config` flags.

Reviewer: ```suggestion
**Note:** The collector can use multiple config files by specifying multiple `--config` flags.
```
In the Note, explain when the user might need to use multiple config files.

Author: Removed this line.
